### DISCUSSION TITLE
Attribute-based mapping with ReverseMap and Ignore

### DISCUSSION THREAD

**Discussion by timmi-on-rails** [LuckyPennySoftware/AutoMapper#3712] [Ideas]
### Source/destination types

```csharp
class Blog
{
    public int A { get; set; }
}

[AutoMap(typeof(Blog), ReverseMap = true)]
class BlogDto
{
    [Ignore]
    public int A {  get; set; }
}
```

I would expect that the `[Ignore]` causes the member to be ignored in both directions.
But instead it is ignored, when mapping `Blog` to `BlogDto`, but it is not ignored in the reverse mapping.
Also I find it confusing, that the `[IgnoreMap]` attribute works, like I expect the `[Ignore]` to work.

This comment here shows me, that the `[Ignore]` should work both ways, because the issuer wanted a one way solution.
https://github.com/AutoMapper/AutoMapper/issues/3362#issuecomment-627384152

**Comment by lbargaoanu** [comment]
`ReverseMap` only reverses `MapFrom`, not `Ignore`. But you can probably write your own `ReverseMap` that does what you want. See also [this](https://github.com/AutoMapper/AutoMapper/blob/upgrade_guide/docs/11.0-Upgrade-Guide.md#maptoattribute-and-ignoremapattribute-were-removed).

**Comment by timmi-on-rails** [comment]
I see, so this behaviour is as expected.
Also reversing `Ignore` would be a breaking change, so I guess it won't happen, although to me it would be more intuitive to reverse all member annotations, if it makes sense.
Do you agree?

**Reply by lbargaoanu** [reply]
I'd rather not take the breaking change, but @jbogard might disagree.
